pkg/controllers/workspace_controller.go

```diff
@@ -359,6 +403,41 @@ Retry_withdifferentname:
 	return resources.GetNode(ctx, newMachine.Status.NodeName, c.Client)
 }
 
+func (c *WorkspaceReconciler) CreateNodeClaim(ctx context.Context, wObj *kaitov1alpha1.Workspace, nodeOSDiskSize string) (*corev1.Node, error) {
+RetryWithDifferentName:
+	newNodeClaim := nodeclaim.GenerateNodeClaimManifest(ctx, nodeOSDiskSize, wObj)
+
+	if err := nodeclaim.CreateNodeClaim(ctx, newNodeClaim, c.Client); err != nil {
+		if apierrors.IsAlreadyExists(err) {
+			klog.InfoS("There exists a nodeClaim with the same name, retry with a different name", "nodeClaim", klog.KObj(newNodeClaim))
+			goto RetryWithDifferentName
+		} else {
+
+			klog.ErrorS(err, "failed to create nodeClaim", "nodeClaim", newNodeClaim.Name)
+			if updateErr := c.updateStatusConditionIfNotMatch(ctx, wObj, kaitov1alpha1.WorkspaceConditionTypeNodeClaimStatus, metav1.ConditionFalse,
+				"nodeClaimFailedCreation", err.Error()); updateErr != nil {
+				klog.ErrorS(updateErr, "failed to update workspace status", "workspace", klog.KObj(wObj))
+				return nil, updateErr
+			}
```

Comment:
The pattern of printing error, updating status and catching update status error is repeated. Consider making this own function like
func (c *WorkspaceReconciler) handleError(ctx context.Context, wObj *kaitov1alpha1.Workspace, reason string, err error) error {
klog error
update status
klog if any update status error
return err
}